### PR TITLE
tgui

### PR DESCRIPTION
**Pull request by DamianX**
I cannot figure out why NanoUI doesn't work for me sometimes so I decided to port an entirely different UI framework to work around it.
tgui is actually pretty cool and might make people developing UIs not want to hurt themselves.

This PR converts the AI Intellicard to use tgui because I needed something as a proof of concept that this could work at all.

image

TODO:
- [x] figure out the license, most of it seems MIT but probably not all of it is
- [x] implement preferences, migrations and shiet

:cl:
- tweak: The GPS now uses a new UI framework from TG (no longer NanoUI).

### CODE REVIEW

**Comment by GreatBigFailure** [comment]
That looks really spiffy.

**Comment by robosperg** [comment]
is this version 2.0 or 3.0 of tgui?

**Comment by DamianX** [comment]
uhh yes

**Comment by Kammerjunk** [comment]
wouldn't it be easiest to just ask their collabs what the relevant licenses are?
worst comes to worst, the art assets can probably be replaced by our existing ones

**Comment by DamianX** [comment]
~~@deathride58 3 years ago you added /proc/getviewsize https://github.com/tgstation/tgstation/commit/08504355d4f716813c36826d2362ba03e5a066bf which I copied here, do you agree to relicense your code to the GPLv3?~~ I am no longer using getviewsize.

@stylemistake First off, thanks for tgui, it looks like a seriously awesome piece of software, and thanks for relicensing it to MIT.
However: https://github.com/tgstation/tgstation/pull/50969 made it so `code/modules/tgui/*.dm` are MIT while files under `code/modules/tgui/states` were left untouched. I assume this was an oversight. Until those files are relicensed we cannot proceed with this PR, so I would like to hear from you.

**Comment by Kammerjunk** [comment]
well that's a lot more complicated than I thought

File: tgui/packages/tgui/public/tgui.html [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        /:::/    /                               \::/____/              \::::/    /
+        \::/    /                                                        \::/    /
+         \/____/                                                          \/____/
+</div>
+<marquee class="FatalError__header">
+A fatal exception has occurred at 002B:C562F1B7 in TGUI.
+The current application will be terminated.
+Please remain calm. Get to the nearest NTNet workstation
+and send the copy of the following stack trace to:
+www.github.com/tgstation/tgstation. Thank you for your cooperation.
```
Comment (Exxion) [inline tgui/packages/tgui/public/tgui.html]: Might want to change this

**Comment by DamianX** [comment]
@Arkatos1 Will you agree to relicense Intellicard.js to something that is compatible with our GPLv3?

**Comment by DamianX** [comment]
Nevermind my previous comment, Intellicard.js was apparently made by @actioninja, see above

**Comment by DamianX** [comment]
I converted the GPS to tgui by following the tutorial, it was a very nice experience. 
image

The only files that are AGPLv3 at this point are `states/default.dm` and `states/hands.dm`, which contain simple sanity checks, and `Intellicard.js`.
If @stylemistake can re-license those first two and @actioninja the latter, then we're 100% OK. I can also just remove Intellicard.js entirely at this point if he doesn't get back to me.
Either way, I leave this in the hands of whoever reviews this, I'm done here.

**Comment by actioninja** [comment]
All tgui js files are licensed under MIT.
the states should be as well, that they aren't I think is just an error.

**Comment by DamianX** [comment]
**TL;DR**: This PR used to contain AGPLv3 code, now it should all be MIT. The only thing that uses tgui at the moment is the GPS, for which I have made the UI. **I think this can be merged now.**

I removed all files from `tgui/packages/tgui/styles/interfaces/**/*.scss`, they were AGPLv3 and unused anyway.
I removed Intellicard.js and reverted `aicard.dm`, because the former is still AGPLv3.
I removed `hands.dm` and all code from `default.dm`, because they are still AGPLv3.
I relicensed the files that I created, `Gps.js` and `adjacent.dm` to MIT, and added a proper copyright header.
When/if the states and interfaces are relicensed we can port them, but they are no longer blockers for this PR.

**Comment by PJB3005** [comment]
TGUI is completely fucked under Wine so RIP.

**Comment by actioninja** [comment]
It should still more or less work under wine, we went through a lot of trouble for ie8 working at the very least.
